src/code/z_lifemeter.c

```diff
@@ -1,17 +1,435 @@
+    gFactor = sHeartsPrimFactors[0][1] * factorBeating;
+    bFactor = sHeartsPrimFactors[0][2] * factorBeating;
+
+    interfaceCtx->beatingHeartPrim[0] = (u8)(rFactor + 255) & 0xFF;
+    interfaceCtx->beatingHeartPrim[1] = (u8)(gFactor + 70) & 0xFF;
+    interfaceCtx->beatingHeartPrim[2] = (u8)(bFactor + 50) & 0xFF;
+
+    rFactor = sHeartsEnvFactors[0][0] * factorBeating;
+    gFactor = sHeartsEnvFactors[0][1] * factorBeating;
+    bFactor = sHeartsEnvFactors[0][2] * factorBeating;
+
+    if (1) {}
+    ddType = type;
+
+    interfaceCtx->beatingHeartEnv[0] = (u8)(rFactor + 50) & 0xFF;
+    interfaceCtx->beatingHeartEnv[1] = (u8)(gFactor + 40) & 0xFF;
+    interfaceCtx->beatingHeartEnv[2] = (u8)(bFactor + 60) & 0xFF;
+
+    sHeartsDDPrim[0][0] = 255;
+    sHeartsDDPrim[0][1] = 255;
+    sHeartsDDPrim[0][2] = 255;
+
+    sHeartsDDEnv[0][0] = 200;
+    sHeartsDDEnv[0][1] = 0;
+    sHeartsDDEnv[0][2] = 0;
+
+    sHeartsDDPrim[1][0] = sHeartsDDPrimColors[ddType][0];
+    sHeartsDDPrim[1][1] = sHeartsDDPrimColors[ddType][1];
+    sHeartsDDPrim[1][2] = sHeartsDDPrimColors[ddType][2];
+
+    sHeartsDDEnv[1][0] = sHeartsDDEnvColors[ddType][0];
+    sHeartsDDEnv[1][1] = sHeartsDDEnvColors[ddType][1];
+    sHeartsDDEnv[1][2] = sHeartsDDEnvColors[ddType][2];
+
+    rFactor = sHeartsDDPrimFactors[ddType][0] * ddFactor;
+    gFactor = sHeartsDDPrimFactors[ddType][1] * ddFactor;
+    bFactor = sHeartsDDPrimFactors[ddType][2] * ddFactor;
+
+    sBeatingHeartsDDPrim[0] = (u8)(rFactor + 255) & 0xFF;
+    sBeatingHeartsDDPrim[1] = (u8)(gFactor + 255) & 0xFF;
+    sBeatingHeartsDDPrim[2] = (u8)(bFactor + 255) & 0xFF;
+
+    rFactor = sHeartsDDEnvFactors[ddType][0] * ddFactor;
+    gFactor = sHeartsDDEnvFactors[ddType][1] * ddFactor;
+    bFactor = sHeartsDDEnvFactors[ddType][2] * ddFactor;
+
+    sBeatingHeartsDDEnv[0] = (u8)(rFactor + 200) & 0xFF;
+    sBeatingHeartsDDEnv[1] = (u8)(gFactor + 0) & 0xFF;
+    sBeatingHeartsDDEnv[2] = (u8)(bFactor + 0) & 0xFF;
+}
+
+s32 func_80100A80(GlobalContext* globalCtx) {
+    gSaveContext.health = (s16)globalCtx->interfaceCtx.unk_24E;
+    
+    return 1;
+}
+
+s32 func_80100AA0(GlobalContext* globalCtx) {
+    InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
+
+    interfaceCtx->unk_250 = 0x140;
+    interfaceCtx->unk_24E += 0x10;
+    if (globalCtx->interfaceCtx.unk_24E >= gSaveContext.health) {
+        globalCtx->interfaceCtx.unk_24E = gSaveContext.health;
+        return 1;
+    }
+    return 0;
+}
+
+s32 func_80100AF0(GlobalContext* globalCtx) {
+    InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
+    s32 health;
+
+    if (interfaceCtx->unk_250 != 0) {
+        interfaceCtx->unk_250--;
+    } else {
+        interfaceCtx->unk_250 = 0x140;
+        interfaceCtx->unk_24E -= 0x10;
+        if (interfaceCtx->unk_24E <= 0) {
+            interfaceCtx->unk_24E = 0;
+            globalCtx->damagePlayer(globalCtx, -((health = gSaveContext.health) + 1));
```

Comment:
This looks like a fake match. Would this match instead?
```suggestion
            globalCtx->damagePlayer(globalCtx, -(((void)0, gSaveContext.health) + 1));
```